compiler/codegen/mli.cpp

```diff
@@ -429,12 +476,17 @@ std::string MLIContext::genMarshalRoutine(Type* t, bool out) {
 
   //
   // Handle translation of different type classes here. Note that right now
-  // the only things we can translate are primitive scalars.
+  // what we can translate is limited.
   //
   if (isPrimitiveScalar(t)) {
     gen += this->genMarshalBodyPrimitiveScalar(t, out);
   } else if (t == dtStringC) {
     gen += this->genMarshalBodyStringC(t, out);
+  } else if (t->symbol->hasFlag(FLAG_C_PTR_CLASS) &&
+             getDataClassType(t->symbol)->typeInfo() == dtInt[INT_SIZE_8]) {
+    // A different strategy will be needed if we ever intend to support
```

Comment:
I don't think there ever will be a way to support `c_ptr(c_char)` or any sort of `c_ptr` without knowledge that there is a buffer backing the pointer (along with its size).